lib/Notification/NotificationHelper.php

```diff
@@ -157,27 +169,41 @@ public function sendCardAssigned($card, $userId) {
 
 	/**
 	 * Send notifications that a board was shared with a user/group
-	 *
-	 * @param $boardId
-	 * @param Acl $acl
-	 * @throws \InvalidArgumentException
 	 */
-	public function sendBoardShared($boardId, $acl) {
-		$board = $this->getBoard($boardId);
+	public function sendBoardShared(int $boardId, Acl $acl, bool $markAsRead = false): void {
+		try {
+			$board = $this->getBoard($boardId);
+		} catch (Exception $e) {
+			return;
+		}
+
 		if ($acl->getType() === Acl::PERMISSION_TYPE_USER) {
 			$notification = $this->generateBoardShared($board, $acl->getParticipant());
-			$this->notificationManager->notify($notification);
+			if ($markAsRead) {
+				$this->notificationManager->markProcessed($notification);
+			} else {
+				$notification->setDateTime(new DateTime());
+				$this->notificationManager->notify($notification);
+			}
 		}
 		if ($acl->getType() === Acl::PERMISSION_TYPE_GROUP) {
 			$group = $this->groupManager->get($acl->getParticipant());
+			if ($group === null) {
+				return;
+			}
 			foreach ($group->getUsers() as $user) {
 				$notification = $this->generateBoardShared($board, $user->getUID());
-				$this->notificationManager->notify($notification);
+				if ($markAsRead) {
+					$this->notificationManager->markProcessed($notification);
```

Comment:
mention, due date and other notifications are dropped when showing them I guess?

Reply:
Went through them as well now and made sure that they are properly removed on actions that require it.